pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -795,6 +798,10 @@ allChainsLabel:
 			filterChainMatch = &fcm
 			if filterChainMatchOption[id].Protocol == istionetworking.ListenerProtocolHTTP {
 				httpOpts = configgen.buildSidecarInboundHTTPListenerOptsForPortOrUDS(node, pluginParams)
+				if chain.TLSContext != nil && chain.TLSContext.CommonTlsContext != nil {
```

Comment:
This seems a bit workaround fix by removing something that is already generated somewhere else, if the `tcpMxcALPN` is not supposed to appear here, can we fix by not generating it in the first place?

Reply:
This is a special case for Auto scenario and hence couldn't do it at generation time itself...